### PR TITLE
Availability bug fix

### PR DESCRIPTION
**Pull request by bharatmotwani**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

File: fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability;
+
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java]: // Useless comment

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

File: fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailabilitySpec.groovy [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        setup:
+        String missingDimensionPhysicalName = "missing_dimension"
+        String missingDimensionApiName = "missingDimension"
+
+        AvailabilityTest target = new AvailabilityTest()
+        //build map of druid physical columns and their available time intervals
+        target.physicalAvailabity = [
+                column_1 : new SimplifiedIntervalList([ new Interval("2020-01-01/2020-03-01")]),
+                column_2 : new SimplifiedIntervalList([ new Interval("2020-02-01/2020-03-01")]),
+                missingDimensionPhysicalName : new SimplifiedIntervalList(),
```
Comment (efronbs) [inline fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailabilitySpec.groovy]: needs parenthesis around `missingDimensionPhysicalName`. If there are no parenthesis groovy interprets the key value as a string literal and not a variable reference.

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

File: fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import org.joda.time.DateTime;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+// Test case 1: BREAKING CHANGE with: Hook to .org, push events only, require continutous-integration check for master
+// Results: hit deprecated validation endpoint, on travis org, branch built only, notification of failure propagated.
+// Test 2: Comenting out breaking change seeing if results still propagate.
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java]: This is where I'm logging test cases

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/TestDimension.java  6
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ColumnMaskingAvailability.java  1
- fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/AbstractAvailabilityDecorator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5240921&bid=17179662)

**Pull request closed without merging**